src/net.cpp

```diff
@@ -367,7 +367,7 @@ CNode* CConnman::FindNode(const CService& addr)
 
 bool CConnman::AlreadyConnectedToAddress(const CAddress& addr)
 {
-    return FindNode(static_cast<CNetAddr>(addr)) || FindNode(addr.ToStringAddrPort());
+    return FindNode(static_cast<CNetAddr>(addr));
```

Comment:
> we'd want to know if that didn't hold for some reason

Wonder if the code should assert they are the same, for debug builds.

Reply:
The added fuzz test does assert that if addr:port equals then `CNetAddr` must also equal. Could add `Assume()` here, something like:

```cpp
bool CConnman::AlreadyConnectedToAddress(const CAddress& addr)
{
    const bool match{FindNode(static_cast<CNetAddr>(addr))};
    if (!match) {
        Assume(!FindNode(addr.ToStringAddrPort()));
    }
    return match;
}
```

I think that is a kind of an overkill, given the fuzz and unit tests added, but if reviewers want it, then I will add that.

Reply:
I agree this is overkill.